tests/UI/pages/FO/product/index.js

```diff
@@ -361,6 +382,67 @@ class Product extends FOBasePage {
   isDeliveryInformationVisible(page) {
     return this.elementVisible(page, this.deliveryInformationSpan, 1000);
   }
+
+  /**
+   * Add a product review
+   * @param page {Page} Browser tab
+   * @param productReviewData {ProductReviewData} The content of the product review (title, content, rating)
+   * @returns {Promise<boolean>}
+   */
+  async addProductReview(page, productReviewData) {
+    if (await this.getNumberOfComment(page) !== 0) {
+      await page.click(this.notEmptyReviewAddReviewButton);
+    } else {
+      await page.click(this.emptyReviewAddReviewButton);
+    }
+    await this.waitForVisibleSelector(page, this.productReviewModal);
+    await this.setValue(page, this.reviewTitle, productReviewData.reviewTitle);
+    await this.setValue(page, this.reviewTextContent, productReviewData.reviewContent);
+    await page.click(this.reviewRating(productReviewData.reviewRating));
+    await page.click(this.reviewSubmitButton);
+    await page.click(this.closeReviewSentConfirmationModalButton);
+    await page.waitForTimeout(2000);
+    return await this.getAttributeContent(page, this.reviewSentConfirmationModal, 'style') === 'display: none;';
```

Comment:
Is there a possibility to wait for an event or an element and not for a specific time?
For example, I don't know if you need the attribute content, but you can wait for the `reviewSentConfirmationModal` to be hidden and return a successful message insteed.